HPXMLtoOpenStudio/resources/schedules.rb

```diff
@@ -1397,18 +1397,24 @@ def calc_peak_flow_from_daily_gpm(col_name:,
     return peak_flow
   end
 
-  def get_external_file(tmp_schedules_path)
-    if File.exist? tmp_schedules_path
-      @external_file = OpenStudio::Model::ExternalFile::getExternalFile(@model, tmp_schedules_path)
+  def get_external_file
+    if File.exist? @tmp_schedules_path
+      if @external_file.nil?
+        @external_file = OpenStudio::Model::ExternalFile::getExternalFile(@model, @tmp_schedules_path)
+      else
+        # Re-read file contents
+        @external_file.remove
+        @external_file = OpenStudio::Model::ExternalFile::getExternalFile(@model, @tmp_schedules_path)
```

Comment:
If I'm understanding correctly, you are creating the csv file, reading it back in, deleting it, and then recreating it? 🤢 Why is it so complicated? Is it not possible to just add the deadband offset logic into the existing `convert_setpoints` method?

Reply:
This is because the convert_setpoints is called at the creation of schedulefile instance. I thought the deadband temperature inputs are hard to pass at that point, so I try to modify the created schedule file when we’re handling setpoints in the measure. Maybe it’s not true. I will give it a second thought and try to simplify the logic.

Reply:
It's hard for me to imagine why it might be hard to do it the more straightforward way, but it's possible. Let @joseph-robertson or me know if you need help here.

Reply:
@yzhou601 Looks like you figured it out! Much better. 🎉